### PR TITLE
Use pixman for compositing pointer

### PR DESCRIPTION
**Pull request by bwidawsk**
This needs to be in draft until pixman-rs is more stable.

### CODE REVIEW

**Comment by i509VCB** [comment]
Is the rationale here that the gpu upload for cursors which likely can't use a dmabuf fb is slower than copying into a dumb buffer?

**Comment by cmeissl** [comment]
> Is the rationale here that the gpu upload for cursors which likely can't use a dmabuf fb is slower than copying into a dumb buffer?

The issue with the current code is not the upload, we need that as a fallback anyway for drawing the cursor on the primary plane.

The issue we have is that we can not use the gles renderer to draw the cursor on the cursor plane. This is the reason why we currently draw to an offscreen buffer and then copy the offscreen pixels to a gbm buffer for scan-out on the cursor plane. But this is pretty costly, especially downloading the pixels from the gpu. 
A software renderer would solve that by directly drawing to the gbm buffer.

For some cases it would be possible to directly copy the cursor pixels to the buffer, but that would at least exclude scaling and output transformations.

File: Cargo.toml [hunk 1]
```diff
@@ -68,6 +68,7 @@ scan_fmt = { version = "0.2.3", default-features = false }
 encoding_rs = { version = "0.8.33", optional = true }
 profiling = "1.0"
 smallvec = "1.11"
+pixman = { git = "https://github.com/cmeissl/pixman-rs.git", version = "0.1.0", features = ["drm-fourcc"] }
```
Comment (ids1024) [inline Cargo.toml:71]: If the dependency is only used in `backend/drm/compositor`, it should probably be an optional dependency that `backend_drm` requires.

**Comment by bwidawsk** [comment]
I think this PR is going to get closed in favor of using the pixman renderer with special hooks to avoid a copy. I'll keep it open until a new PR supersedes it.

**Pull request closed without merging**